sensor/kubernetes/listener/resources/pod_store.go

```diff
@@ -12,6 +12,28 @@ type PodStore struct {
 	pods map[string]map[string]map[string]*storage.Pod
 }
 
+// ReconcileDelete is called after Sensor reconnects with Central and receives its state hashes.
+// Reconciliacion ensures that Sensor and Central have the same state by checking whether a given resource
+// shall be deleted from Central.
+func (ps *PodStore) ReconcileDelete(resType, resID string, _ uint64) (string, error) {
+	if resType != "Pod" {
+		return "", nil
+	}
+	var pod *storage.Pod
+	for _, p := range ps.GetAll() {
```

Comment:
This unnecessarily cloning **all** pods in the store and then picking only one. These are a lot of allocations we could avoid. Check the implementation of `GetAll`.

My suggestion is to change as follows:
1) Create a new `getAllNoLock` function that does the same thing as `GetAll` but doesn't lock and doesn't clone.
2) In `GetAll`: call `getAllNoLock` while locking and clone results
3) in `ReconcileDelete`: call `getAllNoLock` while locking and return ID if match (no clone needed)